### PR TITLE
31 search functionality

### LINKED ISSUE #31: Search functionality
### Needs
* #25 
* Decide on how we want to query the backend for this - new api endpoint? On enter press or automatic fetch?

### Description
Based on the type dropdown and search string in the search sidepane, get the most relevant results in a dropdown underneath the searchbar that can be clicked to select the node in the graph view.

### To-do
* [ ] Search by id / uri
* [ ] Search by label

### PR DESCRIPTION
**Pull request by ShuaiShao20010902**
## To-do
Based on the type dropdown and search string in the search sidepane, get the most relevant results in a dropdown underneath the searchbar that can be clicked to select the node in the graph view. 

- [x] Search by id / uri
image
- [x] Search by label
image

## Linked issues
Closes #31

### CODE REVIEW

File: src/client/src/components/GraphSearchSidepane.vue [hunk 1]
```diff
@@ -57,14 +132,40 @@ export default {
...
               <option value="rdf">RDF Label</option>
             </select>
           </div>
 
+          <div v-if="errorMessage">
+            <p class="error">{{ errorMessage }}</p>
+          </div>
+
+          <!-- Scrollable search results -->
+          <div v-if="showResults && results.length > 0" class="search-results">
```
Comment (keyman015) [inline src/client/src/components/GraphSearchSidepane.vue:148]: Would it be possible to to make the results window smaller than the search panel? It looks rather cramped as it currently stands.

image

File: src/client/src/components/GraphSearchSidepane.vue [hunk 2]
```diff
@@ -57,14 +132,40 @@ export default {
...
+            <ul>
```
Comment (keyman015) [inline src/client/src/components/GraphSearchSidepane.vue:149]: Can the unordered list indentation be removed to increase the space given to the results. The margin between results is enough to differentiate between items.

**Review by keyman015 — CHANGES_REQUESTED** [review]

File: src/client/src/components/GraphSearchSidepane.vue (hunk 1 above)
Comment (ShuaiShao20010902) [reply src/client/src/components/GraphSearchSidepane.vue:148]: oh, you mean to make the result window smaller than the search bar right?

Comment (keyman015) [reply src/client/src/components/GraphSearchSidepane.vue:148]: No i mean the width of the result popup window should be smaller than the info sidepane's width

Comment (ShuaiShao20010902) [reply src/client/src/components/GraphSearchSidepane.vue:148]: oh, that make sense

Comment (ShuaiShao20010902) [reply src/client/src/components/GraphSearchSidepane.vue:148]: After modify that, I will show the pic here. Thanks

File: src/client/src/components/GraphSearchSidepane.vue (hunk 2 above)
Comment (ShuaiShao20010902) [reply src/client/src/components/GraphSearchSidepane.vue:149]: and I will show the pic that removed the unordered list indentation and increse the space on here

File: src/client/src/components/GraphSearchSidepane.vue (hunk 1 above)
Comment (ShuaiShao20010902) [reply src/client/src/components/GraphSearchSidepane.vue:148]: image
how about that now, should I change it smaller?

Comment (keyman015) [reply src/client/src/components/GraphSearchSidepane.vue:148]: That looks good thanks

**Review by keyman015 — APPROVED** [review]
LGTM

**Review by paulmmaingi — APPROVED** [review]
LGTM. 
As discussed, debounce time can be adjusted based on feedback from client. (search limits too)

**Pull request merged**